app/models/profile.rb

```diff
@@ -80,6 +80,16 @@ def supported_os_versions
     end.flatten.uniq.sort.reverse
   end
 
+  def supported_os_versions_by_os_major
+    benchmarks = self.class.canonical.where(
+      ref_id: ref_id,
+      upstream: false
+    ).joins(:benchmark).pluck('benchmarks.version')
+    SupportedSsg.by_os_major[os_major_version].select { |ssg| benchmarks.include? ssg.version }.map do |ssg|
+      "#{ssg.os_major_version}.#{ssg.os_minor_version}"
+    end.uniq
+  end
```

Comment:
@skateman I spent a good amount of time looking into ways to make a query that's more efficient than calling the `bm_versions` method and this was the best I could come up with.. It is a little faster technically 🥲 and we're not caching anything by doing it this way.